internal/controllers/ipaddressclaim_test.go

```diff
@@ -852,6 +853,124 @@ var _ = Describe("IPAddressClaimReconciler", func() {
 			})
 		})
 	})
+
+	Context("When an existing IPAddress is missing finalizers and owner refs", func() {
+		const poolName = "test-pool"
+
+		BeforeEach(func() {
+			pool := v1alpha1.InClusterIPPool{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      poolName,
+					Namespace: namespace,
+				},
+				Spec: v1alpha1.InClusterIPPoolSpec{
+					First:   "10.0.0.1",
+					Last:    "10.0.0.254",
+					Prefix:  24,
+					Gateway: "10.0.0.2",
+				},
+			}
+			Expect(k8sClient.Create(context.Background(), &pool)).To(Succeed())
+			Eventually(Get(&pool)).Should(Succeed())
+		})
+
+		AfterEach(func() {
+			deleteNamespacedPool(poolName, namespace)
+			deleteClaim("test", namespace)
+		})
+
+		It("should add the owner references and finalizer", func() {
+			claim := clusterv1.IPAddressClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: namespace,
+				},
+				Spec: clusterv1.IPAddressClaimSpec{
+					PoolRef: corev1.TypedLocalObjectReference{
+						APIGroup: pointer.String("ipam.cluster.x-k8s.io"),
+						Kind:     "InClusterIPPool",
+						Name:     poolName,
+					},
+				},
+			}
+
+			addressSpec := clusterv1.IPAddressSpec{
+				ClaimRef: corev1.LocalObjectReference{
+					Name: "test",
+				},
+				PoolRef: corev1.TypedLocalObjectReference{
+					APIGroup: pointer.String("ipam.cluster.x-k8s.io"),
+					Kind:     "InClusterIPPool",
+					Name:     poolName,
+				},
+				Address: "10.0.0.1",
+				Prefix:  24,
+				Gateway: "10.0.0.2",
+			}
+
+			address := clusterv1.IPAddress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: namespace,
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion: "alpha-dummy",
+							Kind:       "dummy-kind",
+							Name:       "dummy-name",
+							UID:        "abc-dummy-123",
+						},
+					},
```

Comment:
Consider adding 2 cases:
1. no owner references
2. unrelated owner refs (which is already present)

Reply:
Added test, resolving.